src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -49,6 +59,72 @@ public async Task PutJavaScript()
 
             await PutInternal(validatedAsAdmin: false);
         }
+        
+        [RavenAction("/databases/*/indexes/test", "POST", AuthorizationStatus.ValidUser, EndpointType.Write, DisableOnCpuCreditsExhaustion = true)]
+        public async Task TestIndex()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var input = await context.ReadForMemoryAsync(RequestBodyStream(), "Input");
+
+                var testIndexParameters = JsonDeserializationServer.TestIndexParameters(input);
+                var testIndexDefinition = testIndexParameters.IndexDefinition;
+                var query = testIndexParameters.Query;
+                
+                if (testIndexParameters.IndexDefinition is null)
+                    throw new ArgumentException("Index must have an 'IndexDefinition' field");
+                
+                if (testIndexDefinition.Type.IsJavaScript() == false)
+                {
+                    // C# index without admin authorization
+                    if (HttpContext.Features.Get<IHttpAuthenticationFeature>() is RavenServer.AuthenticateConnection feature && feature.CanAccess(Database.Name, requireAdmin: true, requireWrite: true) == false)
+                        throw new UnauthorizedAccessException($"Index {testIndexDefinition.Name} is a C# index but was sent without admin authorization, this is not allowed.");
+                }
+
+                if (testIndexDefinition.Name is null)
+                    testIndexDefinition.Name = Guid.NewGuid().ToString("N");
+
+                if (query is null)
+                {
+                    var queryString = $"from index '{testIndexDefinition.Name}'";
+                    var queryObject = new { Query = queryString };
+                    
+                    query = DocumentConventions.DefaultForServer.Serialization.DefaultConverter.ToBlittable(queryObject, context);
+                }
+
+                using (var index = Database.IndexStore.CreateTestIndexFromDefinition(testIndexDefinition, context.DocumentDatabase, context))
+                {
+                    index.Start();
+
+                    using (var tracker = new RequestTimeTracker(HttpContext, Logger, Database, "Query"))
+                    using (var token = CreateTimeLimitedQueryToken())
+                    using (var queryContext = QueryOperationContext.Allocate(Database))
+                    {
+                        var indexQueryServerSide = IndexQueryServerSide.Create(HttpContext, query, Database.QueryMetadataCache, tracker, database: Database);
+
+                        indexQueryServerSide.WaitForNonStaleResults = true;
+
+                        var entries = await index.IndexEntries(indexQueryServerSide, queryContext, true, token);
```

Comment:
I checked that `true` passed here is the value of `ignoreLimit`. Is that correct?

Reply:
Yes, now it is explicitly named to make it clearer